core/templates/dev/head/exploration_editor/main_tab/StateEditor.js

```diff
@@ -77,35 +79,27 @@ oppia.controller('StateEditor', [
         $scope.isInteractionShown = true;
       }
 
+      $scope.$on('externalSave', function() {
+        if ($scope.contentEditorIsOpen) {
+          $scope.saveTextContent();
+        }
+      });
+
       $rootScope.loadingMessage = '';
     };
 
     $scope.openStateContentEditor = function() {
       if (editabilityService.isEditable()) {
-        $scope.contentMemento = angular.copy($scope.content);
+        $scope.contentEditorIsOpen = true;
       }
     };
 
     $scope.saveTextContent = function() {
```

Comment:
I thought for a bit, but couldn't come up with better (more consistent) names without compromising their individual clarity -- so, leaving this for now.